types/react-instantsearch-core/index.d.ts

```diff
@@ -206,12 +201,15 @@ export type Refinement = {
   index: string;
   id: string;
   value: RefinementValue;
-} & ({
-  currentRefinement: string;
-} | {
-  items: Array<{ label: string, value: RefinementValue }>;
-  currentRefinement: string[];
-});
+} & (
+  | {
+      currentRefinement: string;
```

Comment:
So, accepting my own responsibility here in that I wrote these type defs 😆 
This has caused some interesting code to try to consume this, and at least one Stack Overflow question I had to answer.  As it stands you have to check it like this:

```ts
const r: Refinement = ...
if ('items' in r) {
  // do something with r.items
  // r.currentRefinement is a Array
} else {
  // r.items is a compiler error (items does not exist)
  // r.currentRefinement is a string
}
```

While I know it's technically correct, I think usability would be improved if this typedef became more like this:
```ts
& (
  {
    items?: undefined,
    currentRefinement: string
  } | {
    items: Array<{ label: string; value: RefinementValue }>;
    currentRefinement: String[];
  }
)
```

The reason is that a check for `if (r.items) {}` is much more intuitive than a check for `if ('items' in r) { }` for developers unfamiliar with the `in` syntax.  In Typescript strict mode, the effect ends up the same.

Reply:
```suggestion
      items: undefined;
      currentRefinement: string;
```

I don't use the dts very often for React InstantSearch myself, but that does seem to make sense for now.